### PR TITLE
NMS-10504: Minion-capable DhcpMonitor and DhcpDetector

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://issues.opennms.org/browse/NMS-10504

Removed Dhcpd plugin, service and configuration. The configuration is now managed in your poller-configuration.xml for the DhcpMonitor and in your foreign-source detector configuration for the DhcpDetector. Please review and test...

### CODE REVIEW

File: features/dhcpd/src/main/java/org/opennms/features/dhcpd/impl/DhcpdImpl.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+public class DhcpdImpl implements Dhcpd {
+    private static final Logger LOG = LoggerFactory.getLogger(DhcpdImpl.class);
+    private final Set<TransactionImpl> transactions = new HashSet<>();
+    private int xid = new Random().nextInt();
+    private boolean shutdown = false;
+    private final Listener port67Listener = new Listener(this, BOOTP_REQUEST_PORT);
+    private final Listener port68Listener = new Listener(this, BOOTP_REPLY_PORT);
+
+    public DhcpdImpl() {
+        Runtime.getRuntime().addShutdownHook(new Thread(new Runnable() {
```
[the remark this answers is not in the document]
Comment (christianpape) [reply features/dhcpd/src/main/java/org/opennms/features/dhcpd/impl/DhcpdImpl.java]: Changed it to a destroy-method

File: opennms-provision/opennms-detector-dhcp/src/main/java/org/opennms/netmgt/provision/detector/dhcp/client/DhcpClient.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    private final String m_macAddress;
+    private final String m_myIpAddress;
+    private final boolean m_extendedMode;
+    private final boolean m_relayMode;
+    private final String m_requestIpAddress;
+
+    private Dhcpd m_dhcpd;
+
+    public DhcpClient(final String macAddress, final boolean relayMode, final String myIpAddress, final boolean extendedMode, final String requestIpAddress, final int timeout, final int retries, final Dhcpd dhcpd) {
+        this.m_macAddress = macAddress;
```
[the remark this answers is not in the document]
Comment (christianpape) [reply opennms-provision/opennms-detector-dhcp/src/main/java/org/opennms/netmgt/provision/detector/dhcp/client/DhcpClient.java]: Fixed

**Comment by indigo423** [comment]
Created Docker images for testing with tag `24.0.0-0.20190206.nms-10504-b16`

* https://hub.docker.com/r/opennms/horizon-core-web/tags
* https://hub.docker.com/r/opennms/minion/tags

**Comment by RangerRick** [comment]
Since this is a breaking change that removes the old DHCP config file there should either be an upgrade task to stuff the old DHCP config options into the poller config, or a note in `opennms-doc/releasenotes/src/asciidoc/releasenotes/whatsnew.adoc` about breaking changes.

**Comment by indigo423** [comment]
#### Testplan

✅ Test DHCP up/down detection on Horizon core server
✅ Test DHCP up/down detection on Minion
Test DHCP up/down detection on Horizon core server extended mode
Test DHCP up/down detection on Minion extended mode
Test DHCP up/down detection on Horizon core server relay
Test DHCP up/down detection on Minion relay mode


#### Issues during testing

- Documentation is still referencing the old DHCP background daemon

**Comment by indigo423** [comment]
@j-white @christianpape tested function wise and 👍 from me :) 

Thank you @christianpape 🎖

File: opennms-doc/guide-admin/src/asciidoc/text/service-assurance/monitors/DhcpMonitor.adoc [hunk 3]
```diff
@@ -4,84 +4,57 @@
...
-| Class Name     | `org.opennms.protocols.dhcp.monitor.DhcpMonitor`
-| Remote Enabled | false
+| Class Name     | `org.opennms.netmgt.poller.monitors.DhcpMonitor`
+| Remote Enabled | true
 |===
 
 .Service monitor parameters configured in `poller-configuration.xml`
 [options="header, autowidth"]
 |===
 | Parameter        | Description                                                                                    | Required | Default value
```
Comment (indigo423) [inline opennms-doc/guide-admin/src/asciidoc/text/service-assurance/monitors/DhcpMonitor.adoc]: Why keep we an empty table here?

**Pull request merged**